vision_layer/vision_tasks/include/gate.h

```diff
@@ -1,48 +1,19 @@
 #ifndef GATE_TASK_H
 #define GATE_TASK_H
 
-#include "ros/ros.h"
-#include "sensor_msgs/Image.h"
-#include "opencv2/imgproc/imgproc.hpp"
-#include "opencv2/core/core.hpp"
-#include "opencv2/imgproc/imgproc_c.h"
-#include "opencv2/highgui/highgui.hpp"
-#include <cv_bridge/cv_bridge.h>
-#include <image_transport/image_transport.h>
-#include <dynamic_reconfigure/server.h>
-#include <geometry_msgs/PointStamped.h>
-#include <sensor_msgs/image_encodings.h>
-#include <bits/stdc++.h>
-#include <stdlib.h>
-#include <string>
-#include <std_msgs/Bool.h>
-#include <std_msgs/Float32.h>
-#include <boost/thread.hpp> 
+#include "base_class.h"
```

Comment:
Nope. In general, I believe it is a bad practice to include all header files into one file and then not mention them in the other files. Your code loses a bit of readability. Instead only those header files should be added which are required for that code.